crates/starknet-server/src/contract_class_choice.rs

```diff
@@ -12,39 +16,69 @@ pub enum AccountContractClassChoice {
 }
 
 impl AccountContractClassChoice {
-    pub fn get_path(&self) -> &str {
-        match self {
-            AccountContractClassChoice::Cairo0 => CAIRO_0_ACCOUNT_CONTRACT_PATH,
-            AccountContractClassChoice::Cairo1 => CAIRO_1_ACCOUNT_CONTRACT_SIERRA_PATH,
-        }
-    }
-
-    pub fn get_class(&self) -> Result<ContractClass, anyhow::Error> {
-        let contract_class = match self {
-            Self::Cairo0 => ContractClass::Cairo0(Cairo0ContractClass::RawJson(
-                Cairo0Json::raw_json_from_path(self.get_path())?,
-            )),
-            Self::Cairo1 => ContractClass::Cairo1(ContractClass::cairo_1_from_sierra_json_str(
-                std::fs::read_to_string(self.get_path())?.as_str(),
-            )?),
-        };
-        Ok(contract_class)
-    }
-
-    pub fn get_hash(&self) -> Result<Felt, anyhow::Error> {
-        let hash = match self {
+    pub(crate) fn get_class_wrapper(&self) -> Result<AccountClassWrapper, anyhow::Error> {
+        Ok(match self {
             AccountContractClassChoice::Cairo0 => {
-                Cairo0Json::raw_json_from_path(self.get_path())?.generate_hash()?
+                let contract_class = Cairo0ContractClass::RawJson(Cairo0Json::raw_json_from_path(
+                    CAIRO_0_ACCOUNT_CONTRACT_PATH,
+                )?);
+                AccountClassWrapper {
+                    class_hash: contract_class.generate_hash()?,
+                    contract_class: ContractClass::Cairo0(contract_class),
+                }
             }
             AccountContractClassChoice::Cairo1 => {
-                let contract_class_str = std::fs::read_to_string(self.get_path())?;
-                let account_contract_class = ContractClass::Cairo1(
+                let contract_class_str =
+                    std::fs::read_to_string(CAIRO_1_ACCOUNT_CONTRACT_SIERRA_PATH)?;
+                let contract_class = ContractClass::Cairo1(
                     ContractClass::cairo_1_from_sierra_json_str(&contract_class_str)?,
                 );
-                account_contract_class.generate_hash()?
+                AccountClassWrapper { class_hash: contract_class.generate_hash()?, contract_class }
             }
-        };
-        Ok(hash)
+        })
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct AccountClassWrapper {
+    pub contract_class: ContractClass,
+    pub class_hash: Felt,
+}
+
+impl FromStr for AccountClassWrapper {
+    type Err = anyhow::Error;
+
+    fn from_str(path_candidate: &str) -> Result<Self, Self::Err> {
+        // load artifact
+        let contract_class = ContractClass::cairo_1_from_sierra_json_str(
+            std::fs::read_to_string(path_candidate)?.as_str(),
+        )?;
+
+        // check that artifact is really account
```

Comment:
I am not sure if this validation should be here. In my opinion the Starknet::new method have to handle this. The cli validates the format of the arguments

Reply:
In a way, we are validating user input here (not the input string itself, but what the string refers to).

In case we want to move this validation to the proposed later stage, we need `std::process::exit` logic in `Starknet`, but the only exiting that we've had so far (that I'm aware of) is in CLI argument validation (encapsulated in clap).

Reply:
Why there is a need for std::process:exit

Reply:
We can also exit with `panic`, but I want to avoid that. In any case, we need to somehow exit because Devnet cannot proceed with an invalid class. If it did proceed with a class that e.g. doesn't have `__execute__`, this would be spotted only with the first transaction execution that relies on the predeployed account (too late).

Reply:
returning from Starknet::new dont need panicking to stop execution.

Reply:
but looking at how the error is formatted and shown to the user, its looking better as It is now